api/src/convert/tx_out_gift_code.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Convert to/from printable::TxOutGiftCode
+
+use crate::{external, printable, ConversionError};
+use mc_crypto_keys::{CompressedRistrettoPublic, RistrettoPrivate, RistrettoPublic};
+use std::convert::TryFrom;
+
+/// Convert mc_transaction_core::TxOutGiftCode --> printable::TxOutGiftCode.
+impl From<&mc_transaction_core::TxOutGiftCode> for printable::TxOutGiftCode {
+    fn from(src: &mc_transaction_core::TxOutGiftCode) -> Self {
+        let mut tx_out_gift_code = printable::TxOutGiftCode::new();
+        tx_out_gift_code.set_global_index(src.global_index);
+        tx_out_gift_code
+            .set_onetime_private_key(external::RistrettoPrivate::from(&src.onetime_private_key));
+        tx_out_gift_code.set_shared_secret(external::CompressedRistretto::from(&src.shared_secret));
+
+        tx_out_gift_code
+    }
+}
+
+/// Convert from printable::TxOutGiftCode --> mc_transaction_core::TxOutGiftCode
+impl TryFrom<&printable::TxOutGiftCode> for mc_transaction_core::TxOutGiftCode {
+    type Error = ConversionError;
+
+    fn try_from(src: &printable::TxOutGiftCode) -> Result<Self, Self::Error> {
+        let global_index = src.get_global_index();
+        let onetime_private_key = RistrettoPrivate::try_from(src.get_onetime_private_key())?;
+        let compressed_shared_secret = CompressedRistrettoPublic::try_from(src.get_shared_secret())
+            .map_err(|_| ConversionError::KeyCastError)?;
```

Comment:
Why is map_err needed here at all? It appears that the [conversions](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/api/src/convert/compressed_ristretto.rs#L60) are already returning a `ConversionError`?

Reply:
So they are. Removed